### PR TITLE
Feature/recommend internal ids

### PR DESCRIPTION
**Pull request by feldlime**
- Added ability to pass internal ids to `recommend` and `recommend_to_items` methods and get internal ids back
- Fixed bug in `LastNSplitter` with incorrect `i_split` in info

### CODE REVIEW

File: rectools/models/base.py [hunk 1]
```diff
@@ -85,23 +89,32 @@ def recommend(
...
             Whether to add rank column to recommendations.
             If True column `Columns.Rank` will be added.
             This column contain integers from 1 to ``number of user recommendations``.
             In any case recommendations are sorted per rank for every user.
             The lesser the rank the more recommendation is relevant.
+        assume_external_ids : bool, default True
+            When ``True`` all input user and item ids are supposed to be external.
+            Internal otherwise. Works faster with ``False``.
+        return_external_ids : bool, default True
+            When ``True`` user and item ids in returning recommendations dataset will be external.
```
Comment (blondered) [inline rectools/models/base.py:108]:
```suggestion
            When ``True`` user and item ids in returning recommendations table will be external.
```

File: rectools/models/base.py [hunk 2]
```diff
@@ -85,23 +89,32 @@ def recommend(
...
+            Internal otherwise. Works faster with ``False``.
 
         Returns
         -------
         pd.DataFrame
-            Recommendations table with columns `Columns.User`, `Columns.Item`, `Columns.Score`\ [, `Columns.Rank`]\.
-            1st column contains external user ids,
-            2nd - external ids of recommended items sorted for each user by relevance,
+            Recommendations table with columns `Columns.User`, `Columns.Item`, `Columns.Score`[, `Columns.Rank`]\.
+            1st column contains external (internal if `return_external_ids` is ``False``) user ids,
```
Comment (blondered) [inline rectools/models/base.py:115]:
```suggestion
            1st column contains user ids
```

File: rectools/models/base.py [hunk 3]
```diff
@@ -85,23 +89,32 @@ def recommend(
...
+            2nd - external (internal if `return_external_ids` is ``False``) ids of recommended items
```
Comment (blondered) [inline rectools/models/base.py:116]:
```suggestion
            2nd - ids of recommended items sorted by relevance for each user
```

File: rectools/models/base.py [hunk 4]
```diff
@@ -175,23 +190,32 @@ def recommend_to_items(
...
              Whether to add rank column to recommendations.
              If True column `Columns.Rank` will be added.
              This column contain integers from 1 to ``number of item recommendations``.
              In any case recommendations are sorted per rank for every target item.
              Less rank means more relevant recommendation.
+        assume_external_ids : bool, default True
+            When ``True`` all input item ids are supposed to be external.
+            Internal otherwise. Works faster with ``False``.
+        return_external_ids : bool, default True
+            When ``True`` item ids in returning recommendations dataset will be external.
```
Comment (blondered) [inline rectools/models/base.py:209]:
```suggestion
            When ``True`` item ids in returning recommendations table will be external.
```

File: rectools/models/base.py [hunk 5]
```diff
@@ -175,23 +190,32 @@ def recommend_to_items(
...
         Returns
         -------
         pd.DataFrame
-            Recommendations table with columns `Columns.TargetItem`, `Columns.Item`, `Columns.Score`, [,`Columns.Rank`].
-            1st column contains external target item ids,
-            2nd - external ids of recommended items sorted for each target item by relevance,
+            Recommendations table with columns `Columns.TargetItem`, `Columns.Item`, `Columns.Score`[, `Columns.Rank`].
+            1st column contains external (internal if `return_external_ids` is ``False``) target item ids,
+            2nd - external (internal if `return_external_ids` is ``False``) ids of recommended items
+                  sorted for each target item by relevance,
```
Comment (blondered) [inline rectools/models/base.py:218]:
```suggestion
            2nd - ids of recommended items sorted by relevance for each target item
```

File: rectools/models/base.py [hunk 6]
```diff
@@ -175,23 +190,32 @@ def recommend_to_items(
...
+            When ``True`` item ids in returning recommendations dataset will be external.
+            Internal otherwise. Works faster with ``False``.
 
         Returns
         -------
         pd.DataFrame
-            Recommendations table with columns `Columns.TargetItem`, `Columns.Item`, `Columns.Score`, [,`Columns.Rank`].
-            1st column contains external target item ids,
-            2nd - external ids of recommended items sorted for each target item by relevance,
+            Recommendations table with columns `Columns.TargetItem`, `Columns.Item`, `Columns.Score`[, `Columns.Rank`].
```
Comment (blondered) [inline rectools/models/base.py:215]:
```suggestion
            Recommendations table with columns `Columns.TargetItem`, `Columns.Item`, `Columns.Score`[, `Columns.Rank`]. External item ids are used by default. For internal ids set `return_external_ids` to ``False``
```

File: rectools/models/base.py [hunk 7]
```diff
@@ -175,23 +190,32 @@ def recommend_to_items(
...
+            1st column contains external (internal if `return_external_ids` is ``False``) target item ids,
```
Comment (blondered) [inline rectools/models/base.py:216]:
```suggestion
            1st column contains target item ids,
```

File: rectools/models/base.py [hunk 8]
```diff
@@ -85,23 +89,32 @@ def recommend(
...
+            When ``True`` user and item ids in returning recommendations dataset will be external.
+            Internal otherwise. Works faster with ``False``.
 
         Returns
         -------
         pd.DataFrame
-            Recommendations table with columns `Columns.User`, `Columns.Item`, `Columns.Score`\ [, `Columns.Rank`]\.
-            1st column contains external user ids,
-            2nd - external ids of recommended items sorted for each user by relevance,
+            Recommendations table with columns `Columns.User`, `Columns.Item`, `Columns.Score`[, `Columns.Rank`]\.
```
Comment (blondered) [inline rectools/models/base.py:114]:
```suggestion
            Recommendations table with columns `Columns.User`, `Columns.Item`, `Columns.Score`[, `Columns.Rank`]\. External user and item ids are used by default. For internal ids set `return_external_ids` to ``False``
```

**Review by blondered — CHANGES_REQUESTED** [review]

File: rectools/models/base.py (hunk 4 above)
Comment (feldlime) [reply rectools/models/base.py:209]: fixed

File: rectools/models/base.py (hunk 2 above)
Comment (feldlime) [reply rectools/models/base.py:115]: fixed

File: rectools/models/base.py (hunk 3 above)
Comment (feldlime) [reply rectools/models/base.py:116]: fixed

File: rectools/models/base.py (hunk 7 above)
Comment (feldlime) [reply rectools/models/base.py:216]: fixed

File: rectools/models/base.py (hunk 5 above)
Comment (feldlime) [reply rectools/models/base.py:218]: fixed

File: rectools/models/base.py (hunk 8 above)
Comment (feldlime) [reply rectools/models/base.py:114]: fixed

File: CHANGELOG.md [hunk 9]
```diff
@@ -6,6 +6,13 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
 
 
+## Unreleased
```
[the remark this answers is not in the document]
Comment (feldlime) [reply CHANGELOG.md:9]: fixed

**Review by blondered — APPROVED** [review]

**Pull request merged**